docs/kbs_protocol.md

```diff
@@ -0,0 +1,265 @@
+In order to achieve this, the public key of KBS needs to be transmitted to KBC through an effective way, 
+and the specific way of public key distribution is not within the scope of this document.
+In addition, it should be noted that the confidentiality protection provided by HTTPS alone is not enough 
+to meet the security requirements of KBS protocol, therefore, as mentioned above, the key generated 
+by hw-tee needs to be used to encrypt and protect the confidential data.
+
+# RCAR semantics
+
+The semantics of attestation defined by Universal-KBS is a simple and extensible four-step model, 
+which uses JSON structure to organize information. As follows:
+
+1. **Request**: The KBC actively sends to KBS, calls the service API provided by KBS to request resources or services.
+2. **Challenge**: After receiving the request, KBS returns a challenge to KBC and asks KBC to 
+send evidence to prove that its environment (HW-TEE) is safe and reliable.
+3. **Attestation**: KBC sends the evidence and the newly generated public key in HW-TEE to KBS.
+4. **Response**: After the KBS verified the evidence, it returns the API output of the requested 
+service and a token to KBC, where the API output payload is protected by the public key in step 3. 
+Within the valid time of the token, KBC can directly request resources or services of KBS by 
+virtue of the token without step 2 and step 3.
+
+## Request
+
+The payload format of the request is as follows:
+
+(Note that the `/*...*/` comments are not valid in JSON, and must not be used in real message.)
+
+```json
+{
+    "request": {
+	    /* Protocol version number used by KBC */
+	    "protocol-version": "0.1.0",
+	    /* Types of HW-TEE platforms where KBC is located, such as "TDX", "SEV-SNP", etc. */
+	    "tee": "",
+	    /* Access token to avoid multiple attempts triggered by consecutive requests. It can be set to blank. */
+	    "token": "",
+	    /* The service API requested to be called. The supported API types are defined and published by KBS. */
+	    "api": "",
+	    /* API input parameters. */
+	    "api-input": {},
+	    /* Reserved fields are used to support some special requests sent by HW-TEE. */
+	    "extra-params": {}
+	}
+}
+```
+
+- `protocol-version`
+
+The protocol version number supported by KBC. KBS needs to judge whether this KBC can communicate 
+normally according to this field. If the protocol version used by this KBC is not supported by 
+KBS, KBS will directly disconnect the link and ignore this request.
+
+- `tee`
+
+Used to declare the type of HW-TEE platform where KBC is located.
+
+- `token`
+
+If other requests have been made before this request, in the previous response KBS would return a 
+token to KBC. The `token` field in request provides the token. If the token is within the 
+validity period, the communication can skip the Challenge and Attestation stage. If no other 
+request has been made before this request, the 'token' field can be left blank.
+
+- `api`
```

Comment:
Shouldnt this be part of the KBS service API endpoint? In other words, shouldnt e.g. getting a key be defined as a `POST /kbs/v1/key` request with the body being the `Request` payload? Then KBS would respond with a `HTTP 200` where the body would be either be a `Response`or a `Challenge`.

I'm obviously missing something. But what is the fundamental reason for not using HTTP request/response?